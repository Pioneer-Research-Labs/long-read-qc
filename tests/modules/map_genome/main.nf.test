nextflow_process {

    name "Test Process map_genome"
    script "modules/map_genome/main.nf"
    process "map_genome"

    test("Should map sequences to genome") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/5HKJT6_1_sample_1.fastq.gz'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/c.00323.dna'),
                file("s3://pioneer-data/genomes/H_elongata/H_elongata_contigs.fna".toString()))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
            ("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/mapped_genome_stats.tsv"),
            ("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/mapped_genome.bam"),
            ("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/mapped_genome.bam.bai")).match()
        }

    }

}
