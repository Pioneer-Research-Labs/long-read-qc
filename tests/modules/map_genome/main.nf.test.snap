{
    "Should map sequences to genome": {
        "content": [
            "mapped_genome_stats.tsv:md5,e40eb11a0412ae9959d94d6c49dadf90",
            "mapped_genome.bam:md5,92b97f1b99703bd6708565889068cf1c",
            "mapped_genome.bam.bai:md5,2e6c20bf80a0099f4e360f6e3f1e5832"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T12:32:52.515432"
    }
}